x/stakeibc/keeper/icacallbacks_reinvest.go

```diff
@@ -33,50 +36,61 @@ func (k Keeper) UnmarshalReinvestCallbackArgs(ctx sdk.Context, reinvestCallback
+	// Fetch callback args
+	reinvestCallback, err := k.UnmarshalReinvestCallbackArgs(ctx, args)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrUnmarshalFailure, fmt.Sprintf("Unable to unmarshal reinvest callback args: %s", err.Error()))
+	}
+	chainId := reinvestCallback.HostZoneId
+	k.Logger(ctx).Info(utils.LogICACallbackWithHostZone(chainId, ICACallbackID_Reinvest, "Starting reinvest callback"))
+
+	// Check for timeout (ack nil)
+	// No action is necessary on a timeout
 	if ack == nil {
-		// handle timeout
-		k.Logger(ctx).Error(fmt.Sprintf("ReinvestCallback timeout, ack is nil, packet %v", packet))
+		k.Logger(ctx).Error(utils.LogICACallbackWithHostZone(chainId, ICACallbackID_Reinvest,
+			"TIMEOUT (ack is nil), Packet: %+v", packet))
 		return nil
 	}
 
+	// Check for a failed transaction (ack error)
+	// No action is necessary on a failure
 	txMsgData, err := icacallbacks.GetTxMsgData(ctx, *ack, k.Logger(ctx))
 	if err != nil {
-		k.Logger(ctx).Error(fmt.Sprintf("failed to fetch txMsgData, packet %v", packet))
+		k.Logger(ctx).Error(fmt.Sprintf("ReinvestCallback failed to fetch txMsgData, packet %v", packet))
 		return sdkerrors.Wrap(icacallbackstypes.ErrTxMsgData, err.Error())
 	}
-
 	if len(txMsgData.Data) == 0 {
-		// handle tx failure
-		k.Logger(ctx).Error(fmt.Sprintf("ReinvestCallback tx failed, txMsgData is empty, ack error, packet %v", packet))
+		k.Logger(ctx).Error(utils.LogICACallbackWithHostZone(chainId, ICACallbackID_Reinvest,
+			"ICA TX FAILED (ack is empty / ack error), Packet: %+v", packet))
 		return nil
 	}
 
-	// deserialize the args
-	reinvestCallback, err := k.UnmarshalReinvestCallbackArgs(ctx, args)
-	if err != nil {
-		return err
-	}
-	amount := reinvestCallback.ReinvestAmount.Amount
-	denom := reinvestCallback.ReinvestAmount.Denom
+	k.Logger(ctx).Info(utils.LogICACallbackWithHostZone(chainId, ICACallbackID_Reinvest, "SUCCESS, Packet: %+v", packet))
 
-	// fetch epoch
+	// Get the current stride epoch number
 	strideEpochTracker, found := k.GetEpochTracker(ctx, epochtypes.STRIDE_EPOCH)
 	if !found {
 		k.Logger(ctx).Error("failed to find epoch")
 		return sdkerrors.Wrapf(types.ErrInvalidLengthEpochTracker, "no number for epoch (%s)", epochtypes.STRIDE_EPOCH)
 	}
-	epochNumber := strideEpochTracker.EpochNumber
-	// create a new record so that rewards are reinvested
+	currentStrideEpochNumber := strideEpochTracker.EpochNumber
+
+	// Create a new deposit record so that rewards are reinvested
 	record := recordstypes.DepositRecord{
-		Amount:             amount,
-		Denom:              denom,
+		Amount:             reinvestCallback.ReinvestAmount.Amount,
+		Denom:              reinvestCallback.ReinvestAmount.Denom,
 		HostZoneId:         reinvestCallback.HostZoneId,
 		Status:             recordstypes.DepositRecord_DELEGATION_QUEUE,
 		Source:             recordstypes.DepositRecord_WITHDRAWAL_ICA,
-		DepositEpochNumber: epochNumber,
+		DepositEpochNumber: currentStrideEpochNumber,
```

Comment:
why are we referencing attributes directly in 86-87, yet creating a temp var in 82 to be used here?

Reply:
Good eye - that wasn't intentional! I can remove the `currentStrideEpochNumber` temp variable. 
But I do think in that scenario it provides a little extra clarification over using just the attribute (i.e. that we're creating an DR with the _current_ epoch).

I could go either way though! Wdyt?

Reply:
I'd err on the side of consistency, not a big deal either way though so will fully defer to you

Reply:
updated!